File: presentation/src/main/java/will/shiro/desafiopicpay/util/di/module/ActivityBindingModule.java
package will.shiro.desafiopicpay.util.di.module;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import will.shiro.desafiopicpay.util.di.scope.ActivityScope;
import will.shiro.desafiopicpay.view.MainActivity;

@Module
public interface ActivityBindingModule {

    @ActivityScope
    @ContributesAndroidInjector(modules = {FragmentBindingModule.class})
    MainActivity contributeMainActivity();
}

File: presentation/src/main/java/will/shiro/desafiopicpay/util/di/module/ActivityBindingModule.kt
package will.shiro.desafiopicpay.util.di.module

import dagger.Module
import dagger.android.ContributesAndroidInjector
import will.shiro.desafiopicpay.view.MainActivity
import will.shiro.desafiopicpay.util.di.scope.ActivityScope

@Module
interface ActivityBindingModule {

    @ActivityScope
    @ContributesAndroidInjector
    fun contributeMainActivity(): MainActivity
}